Mastodon.API/MastodonApiException.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Net;
+
+namespace Mastodon.API
+{
+    /// <summary>
+    /// This exception indicates an error code from the MastodonAPI (i.e., 400, 404, 500, etc).
+    /// It is up to the caller to catch these exceptions and take appropriate action within their application.
+    /// </summary>
+    public class MastodonApiException : Exception
+    {
+        /// <summary>
+        /// The StatusCode returned from the Http call to the API
+        /// </summary>
+        public HttpStatusCode StatusCode { get; }
+        /// <summary>
+        /// Deserialized representation of the error message from the API
+        /// </summary>
+        public Error Error { get; set; }
```

Comment:
Because I forgot to remove the setter.  I'll take care of that before merging.  Thanks.